Examples/Detectors/Geant4Detector/src/MockupSectorBuilder.cpp

```diff
@@ -0,0 +1,344 @@
+  ActsExamples::GdmlDetectorConstruction geo_gdml(m_cfg.gdml_path);
+  g4World = geo_gdml.Construct();
+}
+
+std::shared_ptr<Acts::Experimental::DetectorVolume>
+ActsExamples::MockupSectorBuilder::BuildChamber(
+    const Acts::GeometryContext& gctx,
+    const ActsExamples::MockupSectorBuilder::ChamberConfig& chamber_config) {
+  if (g4World == nullptr) {
+    throw std::invalid_argument(
+        "MockupSector: No g4World initialized or number ");
+    return nullptr;
+  }
+
+  // Geant4Detector Config creator with the g4world from the gdml file
+  auto g4World_config = ActsExamples::Geant4::Geant4Detector::Config();
+  g4World_config.name = "Chamber";
+  g4World_config.g4World = g4World;
+
+  // Get the sensitive and passive surfaces and pass to the g4World Config
+  auto g4Sensitive =
+      std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(
+          chamber_config.SensitiveNames);
+  auto g4Passive =
+      std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(
+          chamber_config.PassiveNames);
+
+  auto g4SurfaceOptions = Acts::Geant4DetectorSurfaceFactory::Options();
+  g4SurfaceOptions.sensitiveSurfaceSelector = g4Sensitive;
+  g4SurfaceOptions.passiveSurfaceSelector = g4Passive;
+  g4World_config.g4SurfaceOptions = g4SurfaceOptions;
+
+  auto g4detector = ActsExamples::Geant4::Geant4Detector();
+
+  auto [surfaces, detectorElements] =
+      g4detector.convertGeant4Volumes(g4World_config);
+
+  // The vector that holds the converted sensitive surfaces of the chamber
+  std::vector<std::shared_ptr<Acts::Surface>> straw_surfaces = {};
+  // The y and z coordinates of the straws' centers
+  std::vector<float> positionY = {};
+  std::vector<float> positionZ = {};
+  std::vector<float> positionX = {};
+
+  // Convert the physical volumes of the detector elements to straw surfaces
+  for (auto& detectorElement : detectorElements) {
+    auto context = Acts::GeometryContext();
+    auto g4conv = Acts::Geant4PhysicalVolumeConverter();
+
+    g4conv.forcedType = Acts::Surface::SurfaceType::Straw;
+    auto g4conv_surf = g4conv.Geant4PhysicalVolumeConverter::surface(
+        detectorElement->g4PhysicalVolume(),
+        detectorElement->transform(context));
+
+    straw_surfaces.push_back(g4conv_surf);
+    positionY.push_back(g4conv_surf->center(context).y());
+    positionZ.push_back(g4conv_surf->center(context).z());
+    positionX.push_back(g4conv_surf->center(context).x());
+  }
+
+  // Create the bounds of the detector volumes
+  float radius = straw_surfaces.front()->bounds().values()[0];
+  auto minY = *min_element(positionY.begin(), positionY.end());
+  auto maxY = *max_element(positionY.begin(), positionY.end());
+
+  auto minZ = *min_element(positionZ.begin(), positionZ.end());
+  auto maxZ = *max_element(positionZ.begin(), positionZ.end());
+
+  auto minX = *min_element(positionX.begin(), positionX.end());
+  auto maxX = *max_element(positionX.begin(), positionX.end());
```

Comment:
You can set these parameters while looping on the straws above. Other thing you can use the `minmax_element` to find the smallest and greatest element in a range.